### PR TITLE
rewrite express and router plugins with new plugin system

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Convert express and router plugins to the new plugin system.

### Motivation

We're moving all plugins so that we can delete the old instrumenter.

### Additional Notes

Supersedes https://github.com/DataDog/dd-trace-js/pull/1940 as @khanayan123 is now working on other things.

### CODE REVIEW

File: packages/datadog-plugin-express/test/index.spec.js [hunk 1]
```diff
@@ -856,6 +854,36 @@ describe('Plugin', () => {
...
+              axios
+                .get(`http://localhost:${port}/user`, {
+                  validateStatus: status => status === 500
+                })
+                .catch(done)
+            })
+          })
+        })
+
+        it('should handle middleware exceptions', done => {
```
Comment (bengl) [inline packages/datadog-plugin-express/test/index.spec.js:883]: What's the difference between this and the previous one.

File: packages/dd-trace/test/plugins/util/web.spec.js [hunk 2]
```diff
@@ -751,138 +749,7 @@ describe('plugins/util/web', () => {
...
-        sockets.forEach(socket => socket.destroy())
-      })
-
-      server.on('connection', (socket) => {
-        sockets.push(socket)
-      })
-
-      const req = http.get(`http://127.0.0.1:${port}`)
-      req.on('error', () => {})
+      expect(tags).to.have.property(ERROR, false)
```
Comment (bengl) [inline packages/dd-trace/test/plugins/util/web.spec.js:752]: Removing all the tests here because they're effectively duplicates?

**Review by bengl — CHANGES_REQUESTED** [review]

File: packages/datadog-plugin-express/test/index.spec.js (hunk 1 above)
Comment (rochdev) [reply packages/datadog-plugin-express/test/index.spec.js:883]: Throwing doesn't call `next`, so it's not the same code path to `throw error` or `next(error)`.

File: packages/dd-trace/test/plugins/util/web.spec.js (hunk 2 above)
Comment (rochdev) [reply packages/dd-trace/test/plugins/util/web.spec.js:752]: Yes, they were basically testing Express more than anything.

File: packages/datadog-plugin-express/test/index.spec.js (hunk 1 above)
Comment (bengl) [reply packages/datadog-plugin-express/test/index.spec.js:883]: Got it :+1:

**Review by bengl — APPROVED** [review]

**Review by bengl — APPROVED** [review]

**Comment by zekth** [comment]
Seems like this change is kind of breaking our current implementation. I'm especially thinking of this line: https://github.com/DataDog/dd-trace-js/pull/1995/files#diff-7266643bfaa9bcb12fcdc9feafd20b7c5696cfc127fbd9c02df36eb9813cc37dR33

In our case we instrument the app using `express` plugin and we have another middleware creating an `AsyncLocalStorage` for business cases. With this upgrade now when we call the `getStore()` it ends up being undefined. @rochdev do you have any lead on this?

Also i stumbled upon this nodejs issue: https://github.com/nodejs/node/issues/41285
I wonder if those are related.

**Comment by rochdev** [comment]
@zekth We're looking into this, but in the meantime I posted a workaround in https://github.com/DataDog/dd-trace-js/issues/2056#issuecomment-1127837699.

**Comment by rochdev** [comment]
@zekth This is now fixed in #2062 and will be in the next release.

**Pull request merged**